frontend/src/navbar.tsx

```diff
@@ -1,27 +1,49 @@
 import { usePulls, useSetFilter } from './pulldasher/pulls-context';
 import { Pull } from './pull';
-import { Center, Flex, Box, BoxProps, Input } from "@chakra-ui/react"
+import { Button, HStack, Center, Flex, Box, BoxProps, Input } from "@chakra-ui/react";
+import { useBoolUrlState } from "./use-url-state";
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faIcicles } from '@fortawesome/free-solid-svg-icons'
 
 export function Navbar(props: BoxProps) {
    const pulls: Pull[] = usePulls();
    const setPullFilter = useSetFilter();
-   const updateFilter = function(event: React.ChangeEvent<HTMLInputElement>) {
+   const [showCryo, setShowCryo] = useBoolUrlState('cryo', false);
+   const updateSearchFilter = function(event: React.ChangeEvent<HTMLInputElement>) {
       const patterns = event.target.value
          .trim()
          .split(/\s+/)
          .filter((s) => s.length)
          .map((s) => new RegExp(s, 'i'));
-      setPullFilter(patterns.length ? (pull: Pull) => {
+      setPullFilter('search', patterns.length ? (pull: Pull) => {
          return patterns.every((pattern) => pull.title.match(pattern));
       }: null);
    };
+   const toggleCryoFilter = function() {
+      setShowCryo(!showCryo);
+      setPullFilter('cryo', showCryo ? null : (pull: Pull) => {
+         return !pull.getLabel("Cryogenic Storage");
+      });
```

Comment:
Would it make sense to pull the `setPullFilter` out of the `toggleCryoFilter`, so we always set the `cryo` filter, but normally set it to `null`? That would eliminate the need for the backwards-ternary here, where we remove the filter if `showCryo` is `true`, and it would make the value of `showCryo` the thing which controls whether or not the filter is present (right now, the filter isn't triggered by the value of `showCryo` but by this toggle function).

Reply:
Sorry... I really don't understand what you're saying. We can't really pull `setPullFilter()` out without causing an infinite loop (calling a `setState()` on every render). I suppose we could nest it in `useEffect()` with `showCryo` as a dependency... but I'm not sure of the point, that'd .....

Oh wait... I see the backward ternary you're talking about, because there's a bug! Because we do `setPullFilter()` in here, it acts on the now-out-of-date value of `showCrypto`. Fixing.

Reply:
Fixed. Good advice.